airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py

```diff
@@ -69,6 +69,27 @@ def time_interval(self) -> dict:
         """
         pass
 
+    def _get_date_time_items_from_schema(self):
+        """
+        Get all properties from schema with format: 'date-time'
+        """
+        result = []
+        schema = self.get_json_schema()["properties"]
+        for key, value in schema.items():
+            if value.get("format") == "date-time":
```

Comment:
Specifically for `source-amplitude` there is no need to dive into nested objects, since the validation checks out only the fields with declared `format: "date-time"`. There are no nested objects with such marks inside the `events` stream schema.